### PR TITLE
Input and display children

### PR DESCRIPTION
**Pull request by chennisden**
Write some api routes such and do some frontend work so that

1) you may now upload children to listings
image

2) each child of a listing is displayed
image

### CODE REVIEW

**Comment by chennisden** [comment]
By the way, this is stacked on top of #202, just because I dont want to deal with potential merge conflict

File: store/app/crud/artifacts.py [hunk 1]
```diff
@@ -109,7 +113,7 @@ async def upload_mjcf(
     ) -> Artifact:
         artifact = Artifact.create(
             user_id=user_id,
-            artifact_type="urdf",
+            artifact_type="mjcf",
```
Comment (codekansas) [inline store/app/crud/artifacts.py:116]: good catch

File: store/app/routers/listings.py [hunk 2]
```diff
@@ -35,6 +35,21 @@ async def list_listings(
     return await crud.get_listings(page, search_query=search_query)
 
 
+@listings_router.get("/batch")
+async def batch(
```
Comment (codekansas) [inline store/app/routers/listings.py:39]: nit: for function names, prefer doing something like `<verb>_<something>`, e.g., `get_batch`

File: store/app/routers/urdf.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    if file.content_type != "application/gzip":
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Only tar.gz files containing a URDF are supported",
+        )
+    artifact = await crud.upload_urdf(
+        file=file.file,
+        user_id=user.id,
+    )
+    return UserInfoResponse(urdf_id=artifact.id)
```
Comment (codekansas) [inline store/app/routers/urdf.py:40]: question - it seems like this lets us upload multiple urdfs per listing, but then the listing endpoint only returns a single urdf?

File: store/app/crud/listings.py [hunk 4]
```diff
@@ -50,3 +53,19 @@ async def add_tag(self, listing_id: str, tag: str) -> None:
...
+        listing = await self._get_item(listing_id, Listing, throw_if_missing=True)
+        artifacts = await self._get_item_batch(listing.artifact_ids, Artifact)
+        urdfs = [artifact for artifact in artifacts if artifact.artifact_type == "urdf"]
+        if len(urdfs) == 0:
+            return None
+        if len(urdfs) > 1:
+            raise InternalError(
+                f"""More than one URDF found for listing {listing_id}.
+                This is due to incorrect data in the database, likely caused by a bug."""
+            )
```
Comment (codekansas) [inline store/app/crud/listings.py:70]: in the `upload` flow, it seems like there is nothing stopping us from uploading multiple urdfs per listing - we should reflect this in the crud side of things

**Review by codekansas — CHANGES_REQUESTED** [review]

File: store/app/routers/urdf.py (hunk 3 above)
Comment (chennisden) [reply store/app/routers/urdf.py]: now only returns the *latest* urdf

**Pull request merged**